packages/component-library/src/components/plmg-text-input/test/plmg-text-input.spec.tsx

```diff
@@ -0,0 +1,145 @@
+import { newSpecPage } from '@stencil/core/testing';
+import { TextInput } from '../plmg-text-input';
+
+describe('plmg-text-input', () => {
+  it('renders default', async () => {
+    const page = await newSpecPage({
+      components: [TextInput],
+      html: `<plmg-text-input label="Default"></plmg-text-input>`,
+    });
+    expect(page.root).toEqualHtml(`
+
+    <plmg-text-input label="Default">
+    <div class="plmg-text-input-wrapper">
+    <label class="medium plmg-text-input-label" htmlfor="default">
+    Default
+    </label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+        <input class="medium" id="default" type="text" name="Default">
+      </div>
+    </plmg-text-input>
+
+    `);
+  });
+  it('renders large', async () => {
+    const page = await newSpecPage({
+      components: [TextInput],
+      html: `<plmg-text-input size="large" label="Large"></plmg-text-input>`,
+    });
+    expect(page.root).toEqualHtml(`
+
+    <plmg-text-input label="Large" size="large">
+    <div class="plmg-text-input-wrapper">
+    <label class="large plmg-text-input-label" htmlfor="large">
+    Large
+    </label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+        <input class="large" id="large" type="text" name="Large">
+        </div>
+    </plmg-text-input>
+    `);
+  });
+  it('renders large', async () => {
+    const page = await newSpecPage({
+      components: [TextInput],
+      html: `<plmg-text-input size="small" label="small"></plmg-text-input>`,
+    });
+    expect(page.root).toEqualHtml(`
+
+    <plmg-text-input label="small" size="small">
+    <div class="plmg-text-input-wrapper">
+    <label class="small plmg-text-input-label" htmlfor="small">
+    small
+    </label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+        <input class="small" id="small" type="text" name="small">
+        </div>
+    </plmg-text-input>
+    `);
+  });
+  it('renders labeless', async () => {
```

Comment:
Confusing test name. Maybe:
```suggestion
  it('renders with the label hidden', async () => {
```